namer/namer.cc

```diff
@@ -1414,6 +1414,57 @@ class SymbolDefiner {
         }
     }
 
+    void deleteViaFullNameHash(core::MutableContext ctx, const core::FoundFieldHash &oldDefHash) {
+        auto ownerRef = core::FoundDefinitionRef(core::FoundDefinitionRef::Kind::Class, oldDefHash.owner.idx);
+        ENFORCE(oldDefHash.nameHash.isDefined(), "Can't delete rename if old hash is not defined");
+
+        // Because a change to classes would have take the slow path, should be safe
+        // to look up old owner in current foundDefs.
+        auto ownerSymbol = getOwnerSymbol(ownerRef);
+        ENFORCE(ownerSymbol.isClassOrModule());
+        auto owner = ownerSymbol.asClassOrModuleRef();
+        if (oldDefHash.owner.onSingletonClass) {
+            owner = owner.data(ctx)->singletonClass(ctx);
+        }
```

Comment:
iiuc, we don't use `methodOwner` here like we do in the other `delete` method because of how the `FoundFieldHash`'s owner field is populated? Or is there some other reason why this snippet does not mirror the corresponding section of the other method?